packages/rulesets/src/spectral/test/properties-type-object-no-definition.test.ts

```diff
@@ -0,0 +1,153 @@
+import { Spectral } from "@stoplight/spectral-core"
+import linterForRule from "./utils"
+let linter: Spectral
+const errorMessage =
+  "If Properties with type:object dont have a reference model defined, then the allowed types can only be primitive data types."
+beforeAll(async () => {
+  linter = await linterForRule("PropertiesTypeObjectNoDefinition")
+  return linter
+})
+test("PropertiesTypeObjectNoDefinition should find errors", () => {
+  const oasDoc = {
+    swagger: "2.0",
+    definitions: {
+      This: {
+        description: "This",
+        type: "object",
+        additionalProperties: {
+          type: "",
+        },
+        tags: {
+          type: "object",
+          additionalProperties: {
+            type: "object",
+            info: {
+              type: "string",
+            },
+          },
+        },
+        properties: {
+          type: "object",
+        },
```

Comment:
Can u align these examples to represent a typical swagger pl?

Reply:
Look at the structure in a real swagger and try and follow that eg: [https://github.com/Azure/azure-rest-api-specs/blob/main/specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json)

Reply:
Typically when a definition is written, the first thing that appears is a properties object which then contains the actual properties.

Reply:
Sure, let me add